app/services/subscription_updator.rb

```diff
@@ -0,0 +1,78 @@
+class SubscriptionUpdator
+    class Error < StandardError; end
+    include Ak::Client
+  
+    def self.run(params)
+      new(params).run
+    end
+  
+    def initialize(params)
+      @params = params[:params]
+      @subscription_id = @params[:recurring_id] || raise(ArgumentError)
+      @transaction_id = @params[:trans_id]
+      @amount = @params[:amount]&.to_d || 0
+    end
+  
+    def run
+      handle_subscription_charge()
+    end
+  
+    private
+  
+    def handle_subscription_charge()
+        return unless subscription
+        # Return if a transaction record already exists for this subscription charge. This is to prevent processing
+        # the same subscription charge twice.
+        return unless Payment::Braintree::Transaction.find_by(transaction_id: @transaction_id).blank?
+
+        update_subscription()
+        create_subscription_charge()
+        true
+      end
+
+      def subscription
+        @subscription ||= Payment::Braintree::Subscription.find_by(subscription_id: @subscription_id)
+      end
+
+      def update_subscription()
+        if @amount > 0 && subscription.amount != @amount
+          subscription.update!(amount: @amount)
+          update_recurring_payment()
+        end
+      end
+
+      def create_subscription_charge()
+        record = Payment::Braintree::Transaction.create!(
+          transaction_id: @transaction_id,
+          subscription: subscription,
+          page: subscription.action.page,
+          customer: customer,
+          status: 'success',
+          amount: @amount,
+          processor_response_code: '1000', #transaction.processor_response_code, #ASkOmar -- can we add this to the msg we send on the webhook listener?
```

Comment:
@osahyoun I hardcoded the value here because I have a question:

The way Champaign [is handling this value now,](https://github.com/SumOfUs/Champaign/blob/development/app/lib/payment_processor/braintree/webhook_handler.rb#L121) it's [by getting it from Braintree's notification](https://github.com/SumOfUs/Champaign/blob/development/app/lib/payment_processor/braintree/webhook_handler.rb#L100-L102), but in the processor, we only have the sqs message available and [it doesn't have this value on it](https://github.com/SumOfUs/Champaign/blob/f9f99bfb7b7c67d4cf8967c8b4d5710af0fbf6e6/app/models/payment/braintree/transaction.rb#L55-L67).
My question is, can we send the `processor_response_code` value from the webhook listener to the queue and then grab it from there in the processor? or are we strictly tied to the current message's format?